packages/core/react-loosely-lazy/src/profiler/index.ts

```diff
@@ -0,0 +1,25 @@
+import { createContext } from 'react';
+
+export type EventInfo = {
+  identifier: string;
+};
+
+export type Profiler = {
+  onLoadStart?(info: EventInfo): void;
+  onLoadComplete?(info: EventInfo): void;
+};
+
+let globalInstance: Profiler | undefined;
+export function setGlobalReactLooselyLazyProfilerInstance(instance: Profiler) {
+  globalInstance = instance;
+}
+
+const globalInstanceProxy = {
+  onLoadStart: (info: EventInfo) => globalInstance?.onLoadStart?.(info),
+  onLoadComplete: (info: EventInfo) => globalInstance?.onLoadComplete?.(info),
+};
+
+export const GlobalReactLooselyLazyProfiler: Readonly<Profiler> =
+  globalInstanceProxy;
+
+export const ProfilerContext = createContext<Profiler>(globalInstanceProxy);
```

Comment:
Passing this through to the global only leaves us unable to distinguish the two. We should instead keep them separate

Reply:
This is only the default value for the context and acts as a fallback. If someone doesn't set a profiler explicitly using a Provider then it will log to the global instance, whatever that is.